contract_comm/blockchain_parameters/blockchain_parameters.go

```diff
@@ -114,6 +114,19 @@ func GetMinimumVersion(header *types.Header, state vm.StateDB) (*params.VersionI
 	return &params.VersionInfo{Major: version[0].Uint64(), Minor: version[1].Uint64(), Patch: version[2].Uint64()}, nil
 }
 
+func IsMinimumVersionAtLeast(major uint64, minor uint64, patch uint64) (bool, error) {
+	desiredVersion := &params.VersionInfo{Major: major, Minor: minor, Patch: patch}
+	minVersion, err := GetMinimumVersion(nil, nil)
+	if err != nil {
+		if err != errors.ErrRegistryContractNotDeployed && err != errors.ErrSmartContractNotDeployed {
+			log.Debug("Error checking client version", "err", err, "contract", hexutil.Encode(params.BlockchainParametersRegistryId[:]))
+			return false, err
+		}
+		return true, nil
```

Comment:
Returning true here make an assumption about the behavior of the caller, specifically that the check should be defualt pass. This was surprising to me when I first saw the call site. I agree that we should handle the "contract not deployed error here", but could this be refactored to give the caller a choice about what to do when that case arises? Returning `(*bool, error)` or `(gte bool, ok bool, error)` or a specific error that can be checked would be some ways to do that, in which case the return value indicates when the minimum client version is "undefined".